Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/util/ContactDiaryExtensions.kt

```diff
@@ -16,8 +17,12 @@ fun ViewPager2.registerOnPageChangeCallback(cb: (position: Int) -> Unit) {
     })
 }
 
-// According to tech spec german locale only
-fun LocalDate.toFormattedDay(): String = toString("EEEE, dd.MM.yy", Locale.GERMAN)
+fun LocalDate.toFormattedDay(): String {
+    // Use two different methods to get the final date format (Weekday, Shortdate)
+    // because the custom pattern of toString() does not localize characters like "/" or "."
+    return "${toString("EEEE", Locale.getDefault())}, " +
+        DateTimeFormat.shortDate().withLocale(Locale.getDefault()).print(this)
```

Comment:
Locale via context instead of static?

Reply:
Align with #2006 which also changes this:

https://github.com/corona-warn-app/cwa-app-android/pull/2006/files#diff-8961af63058c902361512c62b3c1a1764c30f1158d977c682c7e9ca244273c87R21

Reply:
@PhilippNowak96 feel free to tackle this change if you want to :) #2006 also addresses the same changes, but I only included it as a small fix (also not 100% finished yet) which is not directly related to #2006. I will exclude the changes from my PR for now.

Reply:
@axelherbstreith Okay, I will have a look at it later in the evening :)

Reply:
short Note:
according to the current version of the contact diary tech spec, it has to be: withLocale(Locale.Germany...).
"Texts and locale for e.g. date formatting are in German only. "
This is something @PhilippNowak96 you probably can't know in lack of access. :pensive:
I will ask the responsible person for an exception...

Reply:
short Note 2: 
After talk back with our architect, its can stay in this way (locale.getDefault) :blush:
locale.germany only for the export function.
So nothing to complain here from my side

Reply:
@d4rken Good point, thank you! I replaced the static Locale.getDefault() by the already existing SystemInfoProvider and made it using the provided context instead of the system resources. I struggled a little bit with all that Dagger stuff because I didn't deep dive into how you use Dagger and everything related to it yet but I hope I get everything. If anything looks wrong, please let me know! 😃 

@BMItter Thanks for clarifying this. According to #2009 at least iOS already had the "after-fix"-behavior, or do they use a different tech spec? 🤔 Anyways, nice that we can fix this for Android! 😄 

@axelherbstreith Maybe you can adjust your implementation for the accessibility part to also match the visible one and benefit from using the selected locale.